modules/specific-environment-permissions/environment_permissions.tf

```diff
@@ -15,12 +17,20 @@ data "aws_iam_policy_document" "ecs_assume_role" {
       type        = "Service"
       identifiers = ["ecs-tasks.amazonaws.com"]
     }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
   }
 }
 
 resource "aws_iam_role" "jenkins_run_ssm_role" {
```

Comment:
This role can be deleted as it relates to the old Jenkins CI/CD pipeline which has been removed.